src/Raven.Server/Smuggler/Documents/DatabaseSmuggler.cs

```diff
@@ -498,6 +499,20 @@ private SmugglerProgressBase.Counts ProcessRevisionDocuments(SmugglerResult resu
 
                     item.Document.NonPersistentFlags |= NonPersistentDocumentFlags.FromSmuggler;
 
+                    if (item.Document.NonPersistentFlags.Contain(NonPersistentDocumentFlags.FromSmuggler))
```

Comment:
We need to move this inside the `actions.WriteDocument` for DatabaseDestination.

You can throw `DocumentDoesNotExist` exception there, catch it here and increase the errored count.

Reply:
for StreamDestination we don't need it, right?

Reply:
I do not see a reason.